perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java

```diff
@@ -784,11 +785,10 @@ public void checkIfQuotasIsInLimit(Map<String, Pair<BigDecimal, BigDecimal>> quo
 
 			//testing if path is unique
 			String canonicalPath;
-			try {
-				canonicalPath = new URI(path).normalize().getPath();
-				//path should not end on '/' (problem with some systems as GPFS)
-				if(!canonicalPath.equals("/") && canonicalPath.endsWith("/")) canonicalPath = canonicalPath.substring(0, canonicalPath.length() - 1);
-			} catch (URISyntaxException ex) {
+			canonicalPath = Paths.get(path).normalize().toString();
+			//path should not end on '/' (problem with some systems as GPFS)
+			if(!canonicalPath.equals("/") && canonicalPath.endsWith("/")) canonicalPath = canonicalPath.substring(0, canonicalPath.length() - 1);
+			if(!canonicalPath.matches("^[-a-zA-Z.0-9_/:=,]+$")) {
```

Comment:
The regex could be maybe a constant?

Reply:
You are right Metodej, but even better would be to create a static variable outside of this method and precompile the regex there so it does not have to be compiled each time this method Is called.

Reply:
Fixed.